src/frontend/utils.js

```diff
@@ -118,3 +118,65 @@ export const getItemById = (arr, id) => {
   const obj = arr.filter((item) => item.id === id);
   return obj[0];
 };
+
+/**
+ * Converts a camelCase string to its snake_case equivalent
+ * e.g., importantObjectProperty -> important_object_property
+ */
+export const camelCaseToSnakeCase = (s) =>
+  s
+    .split(/(?=[A-Z])/)
+    .join('_')
+    .toLowerCase();
+
+/**
+ * Converts a snake_case string to its camelCase equivalent
+ * e.g., important_object_property -> importantObjectProperty
+ */
+export const snakeCaseToCamelCase = (s) =>
+  s.replace(/[-_][a-z]/g, (group) => group.slice(-1).toUpperCase());
+
+/**
+ * Converts Date object to string in the format '<month>-<day>-<year>'
+ */
+export const dateToString = (date) =>
+  `${date.getMonth() + 1}-${date.getDate()}-${date.getFullYear()}`;
+
+/**
+ * Converts time (represented as Date object) to string in the format '<hour>:<minute> <AM/PM>'
+ */
+export const timeToString = (time) =>
+  time.toLocaleString('en-US', {
+    hour: 'numeric',
+    minute: 'numeric',
+    hour12: true,
+  });
+
+/**
+ * Converts string in the format '<month>-<day>-<year>' to Date Object
+ */
+export const stringToDate = (s) => new Date(s);
+
+/**
+ * Converts string in the format '<hour>:<minute> <AM/PM>' to Date Object
+ * Ref: https://www.freecodecamp.org/news/mathematics-converting-am-pm-to-24-hour-clock
+ */
+export const stringToTime = (s) => {
+  const period = s.slice(-2) === 'AM' ? 'AM' : 'PM';
```

Comment:
This is great, though I think using moment.js might be a better idea here: https://momentjs.com

We have the library installed. All we need to do is specify a format and it'll achieve the same result (and handle some more hidden edge cases)